### DISCUSSION TITLE
Incorrect Tag Author?

### DISCUSSION THREAD

**Discussion by npandelakis** [community/community#23225] [New to GitHub]
<p>Hopefully this is in the correct place, none of the other categories seemed to fit.</p>
<p>I recently added a tag to my repository for a computer science course. Initially, I created, committed and pushed the tag with the wrong date. I then overwrote the tag locally, deleted the tag from GitHub, and pushed my new tag. When I look at the tag however, it says that someone else added the tag, even though that person is not someone I know nor are they a contributor to the repo. Here is the tag in question:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/npandelakis/cs257/releases/tag/hello-world" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:420/420;"><img src="https://avatars3.githubusercontent.com/u/46226654?s=400&amp;amp;v=4" class="thumbnail" width="420" height="420"></div>

<h3><a href="https://github.com/npandelakis/cs257/releases/tag/hello-world" target="_blank" rel="noopener nofollow ugc">Release hello-world · npandelakis/cs257</a></h3>

<p><a href="https://github.com/npandelakis/cs257/releases/tag/hello-world" target="_blank" rel="noopener nofollow ugc">/hello-world</a></p>

  <p><span class="label1">tagged for assignment due 8 January 2021</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Here is the relevant cli history:</p>
<p>1152  git tag -a hello-world -m “tagged for assignment due 20 Sep 2020”<br>
1153  git push origin hello-world<br>
1154  git tag<br>
1155  git tag -a hello-world -m “tagged for assignment due 8 January 2021”<br>
1156  git tag -f hello-world -m “tagged for assignment due 8 January 2021”<br>
1157  git tag<br>
1158  git push origin hello-world<br>
1159  git push origin :refs/tags/hello-world<br>
1160  git push origin hello-world</p>
<p>Anyone have any idea what happened here?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>It looks like something’s wrong with your git settings, specifically the author name and mail. I’m not going to quote the full mail addresses here, but if you look at the commits (<code>git log</code>) and the tag details (<code>git show hello-world</code>) they are all of the form <code>*-MacBook-Pro.local</code>. You need to configure a mail address associated with your Github account as your author email to have them attributed correctly.</p>
<p>Here’s the documentation on how to do that:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address" target="_blank" rel="noopener">Setting your commit email address - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address" target="_blank" rel="noopener">//docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by npandelakis** [comment]
<p>Awesome, thank you for your help! Future tags have me as the author now. Out of curiosity, is the reason Github gets confused because of my very generic admin@Admins-MacBook-Pro.local address?</p>

**Comment by canuckjacq** [comment]
<blockquote>
<p>is the reason Github gets confused because of my very generic admin@Admins-MacBook-Pro.local address?</p>
</blockquote>
<p>Yes - you shouldn’t be using that if you want your commit history to be correctly attributed to you. If you want a private email for contributions, we have a guide for that:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-on-github" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-on-github" target="_blank" rel="noopener">Setting your commit email address - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-on-github" target="_blank" rel="noopener">//docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by airtower-luna** [comment]
<p>I assume so. I’m not sure why it’d lead to the specific account it did (maybe they had the same issue and just added the generic address to their account?), but in general Github uses the author email to detect who wrote what. :slightly_smiling_face:</p>
